torch/_inductor/fx_passes/post_grad.py

```diff
@@ -802,3 +805,162 @@ def repl(inp, x1, x2):
 
     with V.fake_mode:
         match.replace_by_example(repl, [inp, mat1, mat2])
+
+
+def allows_mixed_devices(op):
+    return op in (
+        aten.index.Tensor,
+        aten.index_put.default,
+        aten.index_put_.default,
+        aten.copy.default,
+        aten.copy_.default,
+        aten.slice_scatter.default,
+    )
+
+
+def cannot_be_moved_to_cuda(node):
+    if node.target == "output":
+        return True
+
+    if not isinstance(
+        node.target, torch._ops.OpOverload
+    ) and node.target.namespace not in ("prims", "aten"):
+        return True
```

Comment:
cc @ezyang i'm restricting to aten/prims here but maybe i should be even more selective ? wdyt ?

Reply:
I think the main question is whether or not to allowlist or blocklist at a fine-grained level. It seems... safer to allowlist on a per operator basis? I'm just thinking of situations where it's not OK to move an operator to CUDA and there's lots: (1) the operator is CPU only, (2) the CUDA version has drastically different numerics, (3) the CPU version is faster than the CUDA version / the CUDA version is crappy...

That being said, I think any operator that we have a lowering for is probably OK to move to CUDA. So you can get a decently sized list that way, probably.

Reply:
It feels like we may have to collect some data on this

Reply:
only including operators inductor has lowerings for - i think that should work